Feat/thread side bar

Changyu-Ryou opened the pull request:
## Linked Issue
close #116 
close #117 
close #9 
close #134 

## 공유할 사항
- 스레드 사이드바 구현
- 댓글 추가 및 업데이트 정상 작동
- 리액션 업데이트 사이드 바 내에서 업데이트는 추가하지 않았습니다.

## 논의할 사항
- 없습니다. ❌

Changyu-Ryou committed:
Feat: update backend new reply socket service

백엔드에서 소켓으로 새로운 댓글 생성 기능구현

Changyu-Ryou committed:
Refactor: update getReplyMessages method

Chat model에서 댓글 목록을 조회하는 메소드 수정

Changyu-Ryou committed:
Refactor: update Actionbar component

메시지 type에 따른 Actionbar 렌더링 부 수정

Changyu-Ryou committed:
Feat: add getChatReplyMessage api

프론트에서 댓글 목록을 조회하는 getChatReplyMessage api 추가

Changyu-Ryou committed:
Refactor: update chatMessage

type에 따라 chatMessage가 렌더링하는 부분을 변경하도록 수정

Changyu-Ryou committed:
Refactor: update sidebar on workspacePage

workspacePage에서 chatId의 유무에 따라 우측 사이드바를 표시하도록 변경

Changyu-Ryou committed:
Refactor: remove index switch

index에서 라우팅하는 부분 중 switch 제거

Changyu-Ryou committed:
Refactor: update ThreadReactionCard useEffect input

useEffect에 userInfo를 input으로 추가

Changyu-Ryou committed:
Fix: throttle bug fix

throttle 정상적으로 동작하도록 수정

Changyu-Ryou committed:
Feat: add SideThreadBar component

우측에 댓글 목록을 보여주는 사이드바 컴포넌트 추가

Changyu-Ryou committed:
Refactor: update ChatRoom

messages 없을 경우에 대한 분기처리와 ChatArea의 witdh 수정

bell-won commented on frontend/src/page/WorkspacePage/WorkspacePage.js:
```diff
@@ -1,25 +1,33 @@
-  const { channelId } = useParams()
+  const { channelId, chatId } = useParams()
   const [lineWidth, setLineWidth] = useState(20)
+  const [SideBarWidth, setSideBarWidth] = useState(30)
```
SideBarWidth만 파스칼 케이스인 이유가 있나요?

Changyu-Ryou replied:
이 부분 수정해야겠네욥!

bell-won commented on frontend/src/container/ChatMessage/ChatMessage.js:
```diff
@@ -90,17 +87,20 @@ const ChatMessage = forwardRef(
         {/* TODO view thread reply 구현  */}
-        {reply && reply.length !== 0 && (
-          <ViewThreadBarStyle>view thread</ViewThreadBarStyle>
+        {type !== 'reply' && reply && reply.length !== 0 && (
```
isEmpty 사용하면
```javascript
{type !== 'reply' && !isEmpty(reply) && ...}
```
으로 표현 가능할 것 같습니다.

Changyu-Ryou replied:
넵 isEmpty 적용해 보겠습니다

bell-won commented on frontend/src/container/ChatMessage/ChatMessage.js:
```diff
@@ -61,7 +58,7 @@ const ChatMessage = forwardRef(
     return (
       <StyledMessageContainer
-        type={type}
+        type={type || 'chat'}
```
type에 default 값을 사용하는걸 변경하신 것 같은데
특별한 이유가 있나요 ??

Changyu-Ryou replied:
그러네요 원래 값으로 둬도 괜찮겠네요 이 부분도 다시 수정해보겠습니다.

bell-won reviewed (approved):
고생하셨습니다.

rockpell reviewed (approved):
수고하셨습니다!
공유할 사항에 throttle에 대한 내용도 적어주시면 좋을거 같아요.

Changyu-Ryou committed:
Refactor: update reviewed code

리뷰 받은 코드 수정
- workspacePage useState 파스칼 표기 수정
- chat default 설정 변경
- isEmpty util 사용해 조건 검증
